pkg/apis/seedmanagement/v1alpha1/types_managedseed.go

```diff
@@ -67,12 +67,10 @@ type ManagedSeedSpec struct {
 	// This field is immutable.
 	// +optional
 	Shoot *Shoot `json:"shoot,omitempty" protobuf:"bytes,1,opt,name=shoot"`
-	// SeedTemplate is a template for a Seed object, that should be used to register a given cluster as a Seed.
-	// Either SeedTemplate or Gardenlet must be specified. When Seed is specified, the ManagedSeed controller will not deploy a gardenlet into the cluster
-	// and an existing gardenlet reconciling the new Seed is required.
-	// Deprecated: This field is deprecated and will be removed in a future version of Gardener. Define seed via `.spec.gardenlet.config` instead.
-	// +optional
-	SeedTemplate *gardencorev1beta1.SeedTemplate `json:"seedTemplate,omitempty" protobuf:"bytes,2,opt,name=seedTemplate"`
+
+	// SeedTemplate is tombstoned to show why 2 is reserved protobuf tag.
+	// SeedTemplate *gardencorev1beta1.SeedTemplate `json:"seedTemplate,omitempty" protobuf:"bytes,2,opt,name=seedTemplate"`
```

Comment:
FYI, best practice following this [example](https://github.com/kubernetes/kubernetes/pull/102412/files#diff-5ffbebb40a29f8e491f4ff1e07c618eb3edcc0d5f00a1a71445272d6767d7883L4258). Thanks @ialidzhikov for the reference.

Reply:
Thanks for incorporating my feedback. We could also add an unit test similar to the one added in https://github.com/kubernetes/kubernetes/pull/102412 to make sure that any new field won't reuse the protobuf number of the already removed field.

```go
package v1alpha1_test

import (
	"reflect"
	"strings"

	. "github.com/gardener/gardener/pkg/apis/seedmanagement/v1alpha1"

	. "github.com/onsi/ginkgo/v2"
)

var _ = Describe("ManagedSeed", func() {

	It("should not allow reuse of protobuf number of already removed ManagedSeedSpec fields", func() {
		obj := reflect.ValueOf(ManagedSeedSpec{}).Type()
		for i := 0; i < obj.NumField(); i++ {
			f := obj.Field(i)

			protobufNum := strings.Split(f.Tag.Get("protobuf"), ",")[1]
			if protobufNum == "2" {
				Fail("protobuf 2 in ManagedSeedSpec is reserved for removed seedTemplate field")
			}
		}
	})
})
```

I also have in my list to update the api conventions doc in g/g wrt to this.

Reply:
Thanks @ialidzhikov, added in [dd6510a](https://github.com/gardener/gardener/pull/6972/commits/dd6510aaf6633b76ee12086755d69cdd5373aa30).